### ISSUE TITLE
"Query requires iterating over all vertices" warning when using g.V().hasId(x)

### ISSUE THREAD

**Issue by mikedias** [thinkaurelius/titan#1284]
As discussed in this [topic](https://groups.google.com/forum/#!topic/gremlin-users/Kz2bOeAeqh4), Titan raises a warning when a `g.V().hasId(123)` that has a direct impact on the blvp performance:

`WARN StandardTitanTx  - Query requires iterating over all vertices [(~id = 123)]. For better performance, use indexes.`

Marko suggests that the `TitanGraphStepStrategy` could handle the transformation of `hasId(x)` into `vertices(x)`

**Comment by mikedias** [comment]
The [PR 267](https://github.com/apache/incubator-tinkerpop/pull/267) introduces the method `GraphStep.processHasContainerIds()` that will help to fix this.

**Comment by spmallette** [comment]
@mikedias can we then close this issue over here in titan? or is there more for titan to implement?

**Comment by mikedias** [comment]
@spmallette no. The `TitanGraphStepStrategy` needs to be optimized yet. The PR 267 AFAIK only introduces a helper method to implement the optimization.

**Comment by pluradj** [comment]
https://issues.apache.org/jira/browse/TINKERPOP-1219
